openassessment/xblock/submission_mixin.py

```diff
@@ -250,8 +276,24 @@ def save_files_descriptions(self, data, suffix=''):  # pylint: disable=unused-ar
 
         return {'success': True, 'msg': u''}
 
+    def create_team_submission(self, student_item_dict, student_sub_data):
+        """ A student submitting for a team should generate matching submissions for every member of the team """
+        team_dict = self.get_team_info()
+
```

Comment:
Somewhere in here, you'll want to grab the keys of all files that are shared amongst this team.

Reply:
...and then jam them into each `student_sub_data` dict.

Reply:
This wraps the standard `create_submission` call which I believe already covers that part but please confirm

Reply:
All it does is modifies the `student_item_dict` to change the student IDs

Reply:
That `self.file_manager.get_uploads()` call in `create_submission` doesn't do anything with `SharedFileUploads` yet - we'll have to take care of that in this PR (although it's highly likely that Jansen will need to do very similar work for the next ticket he picks up).

Reply:
That `self.file_manager.get_uploads()` call in `create_submission` doesn't do anything with `SharedFileUploads` yet - we'll have to take care of that in this PR (although it's highly likely that Jansen will need to do very similar work for the next ticket he picks up).